### PR TITLE
701: Fix entry/display of translations that start with a blank line.

### LINKED ISSUE #701: Starting Newline stripped from Translation
Hello,

While translating the iOS app I came across the following entry which had a requirement of the first character being a newline;
screen shot 2017-04-17 at 11 55 25 am
Permalink - https://translate.wordpress.org/projects/apps/ios/dev/en-ca/default?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=4334240&filters%5Btranslation_id%5D=48678505

If you copy the original and submit as a translation the initial newline character is stripped on submit and triggers the warning as the first character is no longer a newline.

Cheers

### PR DESCRIPTION
**Pull request by toolstack**
Resolves #701.

### CODE REVIEW

File: gp-templates/helper-functions.php [hunk 1]
```diff
@@ -41,6 +41,26 @@ function prepare_original( $text ) {
...
+/**
+ * Prepare a translation string to be printed out in a translation row by adding an 'extra' return/newline if
+ * it starts with one.  See GH701.
+ *
+ * @param string $text A single style handle to enqueue or an array or style handles to enqueue.
+ *
+ * @return string The prepared string for output.
+ */
+function prepare_translation_textarea( $text ) {
+	if( gp_startswith( $text, "\r\n" ) ) {
```
Comment (ocean90) [inline gp-templates/helper-functions.php]: Missing whitespace

File: gp-templates/helper-functions.php [hunk 2]
```diff
@@ -41,6 +41,26 @@ function prepare_original( $text ) {
...
+		$text = "\r\n" . $text;
+	} else {
+		if( gp_startswith( $text, "\n" ) ) {
```
Comment (ocean90) [inline gp-templates/helper-functions.php]: Missing whitespace

File: gp-templates/helper-functions.php [hunk 3]
```diff
@@ -41,6 +41,26 @@ function prepare_original( $text ) {
...
 
+/**
+ * Prepare a translation string to be printed out in a translation row by adding an 'extra' return/newline if
+ * it starts with one.  See GH701.
+ *
+ * @param string $text A single style handle to enqueue or an array or style handles to enqueue.
+ *
+ * @return string The prepared string for output.
+ */
+function prepare_translation_textarea( $text ) {
```
Comment (ocean90) [inline gp-templates/helper-functions.php]: Missing `gp_` prefix

File: gp-templates/helper-functions.php [hunk 4]
```diff
@@ -130,7 +150,7 @@ function textareas( $entry, $permissions, $index = 0 ) {
 			</div>
 		<?php endif; ?>
 		<blockquote class="translation"><em><small><?php echo prepare_original( esc_translation( gp_array_get( $entry->translations, $index ) ) ); // WPCS: XSS ok. ?></small></em></blockquote>
-		<textarea class="foreign-text" name="translation[<?php echo esc_attr( $entry->original_id ); ?>][]" id="translation_<?php echo esc_attr( $entry->original_id ); ?>_<?php echo esc_attr( $index ); ?>" <?php echo $disabled; // WPCS: XSS ok. ?>><?php echo esc_translation( gp_array_get( $entry->translations, $index ) ); // WPCS: XSS ok. ?></textarea>
+		<textarea class="foreign-text" name="translation[<?php echo esc_attr( $entry->original_id ); ?>][]" id="translation_<?php echo esc_attr( $entry->original_id ); ?>_<?php echo esc_attr( $index ); ?>" <?php echo $disabled; // WPCS: XSS ok. ?>><?php echo prepare_translation_textarea( esc_translation( gp_array_get( $entry->translations, $index ) ) ); // WPCS: XSS ok. ?></textarea>
```
Comment (ocean90) [inline gp-templates/helper-functions.php]: Shouldn't this be the other way around? `esc_translation( prepare_translation_textarea( … ) )`

**Review by ocean90 — CHANGES_REQUESTED** [review]
Needs an update of the minified file and a version bump.

File: gp-templates/helper-functions.php (hunk 3 above)
Comment (toolstack) [reply gp-templates/helper-functions.php]: Fixed.

File: gp-templates/helper-functions.php (hunk 1 above)
Comment (toolstack) [reply gp-templates/helper-functions.php]: Fixed.

File: gp-templates/helper-functions.php (hunk 2 above)
Comment (toolstack) [reply gp-templates/helper-functions.php]: Fixed.

File: gp-templates/helper-functions.php (hunk 4 above)
Comment (toolstack) [reply gp-templates/helper-functions.php]: Doesn't really matter, I chose this way to make sure the extra `lf` was added as late as possible just in case.

**Comment by toolstack** [comment]
Bumped asset loader version and the minified file is already included in the PR.

**Comment by toolstack** [comment]
The blocking conflict is because we've merged another update to the minification file in to develop and of course git can't resolve the issue.  I'll have to rebase the PR.

**Comment by toolstack** [comment]
PR rebased.

**Review by ocean90 — APPROVED** [review]

**Pull request merged**